### PR TITLE
types: Update id-ctrl field based on NVMe 2.1 spec

### PR DESCRIPTION
**Pull request by sc108-lee**
Boot Partition Capabilities (BPCAP)
NVM Subsystem Shutdown Latency (NSSL)
Power Loss Signaling Information (PLSI)
Controller Reachability Capabilities (CRCAP)
Key Per I/O Capabilities (KPIOC)
Maximum Processing Time for Firmware Activation Without Reset (MPTFAWR)

### CODE REVIEW

File: src/nvme/types.h [hunk 1]
```diff
@@ -2027,6 +2112,19 @@ enum nvme_id_ctrl_anacap {
 	NVME_CTRL_ANACAP_GRPID_MGMT		= 1 << 7,
 };
 
+/**
+ * enum nvme_id_ctrl_kpioc - Key Per I/O Capabilities.
+ * @NVME_CTRL_KPIOC_KPIOS:  If set, then the controller supports the Key Per
+ *			    I/O capability.
+ * @NVME_CTRL_KPIOC_KPIOSC: If set, tthen the Key Per I/O capability applies to
```
Comment (NateThornton) [inline src/nvme/types.h:2119]: nit: `tthen` => `then`

Comment (sc108-lee) [reply src/nvme/types.h:2119]: Thanks for checking

File: src/nvme/types.h [hunk 2]
```diff
@@ -1737,6 +1778,50 @@ enum nvme_id_ctrl_ctratt {
...
+ *					Protection is not specified.
+ *				01b - RPMB Boot Partition Write Protection is
+ *					not supported by this controller.
+ *				10b - RPMB Boot Partition Write Protection is
+ *					supported by this controller.
+ *				11b - Reserved
+ * @NVME_CTRL_BACAP_SFBPWPS: Set Features Boot Partition Write Protection Support
+ */
+enum nvme_id_ctrl_bpcap {
+	NVME_CTRL_BACAP_RPMBBPWPS		= 3 << 0,
```
[the remark this answers is not in the document]
Comment (igaw) [reply src/nvme/types.h:1794]: Yes, please introduce the `_MASK` and `_SHIFT` fields.

File: src/nvme/types.h [hunk 3]
```diff
@@ -1737,6 +1778,50 @@ enum nvme_id_ctrl_ctratt {
...
+/**
+ * enum nvme_id_ctrl_bpcap - Boot Partition Capabilities
+ * @NVME_CTRL_BACAP_RPMBBPWPS: RPMB Boot Partition Write Protection Support.
+ *				00b - Support for RPMB Boot Partition Write
+ *					Protection is not specified.
+ *				01b - RPMB Boot Partition Write Protection is
+ *					not supported by this controller.
+ *				10b - RPMB Boot Partition Write Protection is
+ *					supported by this controller.
+ *				11b - Reserved
```
Comment (igaw) [inline src/nvme/types.h:1790]: Could you also introduce enums for these? Thanks!

**Comment by igaw** [comment]
looks good, just add the mask/shift things and also the missing bit field definition, so that all is consistent. thanks

**Comment by sc108-lee** [comment]
Updated based on review :)

**Comment by igaw** [comment]
Thanks!

**Pull request merged**